### PR TITLE
User tools

### PR DESCRIPTION
**Pull request by GoralTomaszGorski**
- user: automatically email adress adding 
and setNick automatically during addPatient function
- Templates reduced
- Improved style and view, Services
- DeletePatientException one more fixed different way.

### CODE REVIEW

File: src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java [hunk 1]
```diff
@@ -12,49 +14,68 @@
...
-        patientToSave.setNick(patientDto.getNick());
+        if (patientDto.getNick().isEmpty()) {
+            StringBuilder sB = new StringBuilder();
+            String tel = patientDto.getTelephone();
+            int n = tel.length();
+            for (int i = (n - 3);
+                 i < n;
+                 i++) {
+                char c = patientDto.getTelephone().charAt(i);
+                sB.append(c);
```
Comment (gorskimarcin96) [inline src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java:55]: we can replace lines 49-56 to one line:

```java
tel.substring(tel.length() - 3)
```

**Review by gorskimarcin96 — APPROVED** [review]

File: src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java (hunk 1 above)
Comment (GoralTomaszGorski) [reply src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java:55]: how you find it??

Comment (gorskimarcin96) [reply src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java:55]: image
this is not easy... :stuck_out_tongue_winking_eye:

Comment (GoralTomaszGorski) [reply src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java:55]: I saw this answer today, I will check it

Comment (GoralTomaszGorski) [reply src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java:55]: I tried to fix with using above code it is probably better but something wrong when I use substring method.
I fixed set free and true calendar therm.

Comment (GoralTomaszGorski) [reply src/main/java/goral/psychotherapistoffice/domain/patient/PatientService.java:55]: I fixed boolean set free false and true calendar therm during edit and add therm*

**Pull request merged**